src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -8530,7 +8531,7 @@ instruct MoveF2I_stack_reg(iRegINoSp dst, stackSlotF src) %{
 
   effect(DEF dst, USE src);
 
-  ins_cost(LOAD_COST);
+  ins_cost(ALU_COST + LOAD_COST);
```

Comment:
Adding an extra cost of `ALU_COST` for these load/store nodes looks a bit strange to me. I suppose only lowering the cost for those fmv.x.w/fmv.w.x/fmv.x.d/fmv.d.x nodes will do?

Reply:
those nodes need to go below 100 which then starts looking ugly

Reply:
Seems that the performance gain is still there when reverting part of the changes. I haven't checked the JIT code though.

[addon-change.diff.txt](https://github.com/openjdk/jdk/files/13815870/addon-change.diff.txt)

Reply:
Thanks, reverting some changes still leaves good generation. I have performed some more benchmarks on thead board, in all cases necessary instructions are generated in JIT code.